src/control/lib/control/auto.go

```diff
@@ -973,7 +1030,13 @@ func getBdevTiers(log logging.Logger, scmCls storage.Class, ssds *hardware.PCIAd
 
 type newEngineCfgFn func(int) *engine.Config
 
-func genEngineConfigs(log logging.Logger, scmOnly bool, newEngineCfg newEngineCfgFn, nodeSet []int, nd *networkDetails, sd *storageDetails) ([]*engine.Config, error) {
+func genEngineConfigs(req ConfGenerateReq, newEngineCfg newEngineCfgFn, nodeSet []int, nd *networkDetails, sd *storageDetails) ([]*engine.Config, error) {
+	nrFabPorts := len(req.FabricPorts)
+	if nrFabPorts > 0 && nrFabPorts < len(nodeSet) {
+		return nil, errors.Errorf("insufficient fabric ports for nr engines, want %d got %d",
+			len(nodeSet), nrFabPorts)
+	}
```

Comment:
I think we should accept the case where the user passes in a single port that will be used for both engines, if the engines are on different interfaces.

Reply:
apparently this should not be allowed according to @Michael-Hennecke

Reply:
we have seen failures many times when the port numbers of the two engine interfaces are identical, even when the fabric interfaces are different. i do not understand the details _why_ that is, but i can attest _that_ it happens. 

and more pressing than the changes here would be a clear guidance how big the _range_ of ports used by an engine actually is. saying "A difference of ~1000 is recommended." isn't really a satisfactory answer because we do not explain how we arrive at that estimate. on which other config parameters or cluster size metrics does this depend? rather than a single port number, i would actually prefer an interval of port numbers in the fabric_iface_port statement, and then the code can check how many ports it actually needs and fail if the interval is too small.

Reply:
I have requested clarification on this guidance and the rational behind it from @frostedcmos . The ask for actual port number validation and interval range in the config is nontrivial and the work would require scheduling so please create a ticket if you want this feature in 2.6 (or another) release @Michael-Hennecke . In terms of what changes you require to this PR specifically is it an expansion of the port number difference statement to include some detail on how to calculate the interval? or should I just remove that statement for the moment and follow up with a subsequent PR after I have found some more definitive explanation?

Reply:
no change requested from me as this point - yes let's move the "interval" topic to a separate ticket.

Reply:
I've reverted port validation changes in this PR, @Michael-Hennecke is going to discuss the issue of identical port numbers across engines with different interfaces with @frostedcmos and we will address those changes separately.

Reply:
OK with me. Thanks for the explanation.